docs/03-move/100-move-prover/02-mvp-tutorial.md

```diff
@@ -1,5 +1,606 @@
+All built-in integer types (`u8`, `u64`, etc.) are automatically converted to type `num` when referencing data in a Move program in MSL.
+A more detailed description of the type system can be found in the MSL documentation.
+
+## Verify the `transfer` function
+
+The signature of the function `transfer` is as follows:
+
+```rust
+public fun transfer<CoinType: drop>(from: &signer, to: address, amount: u64, _witness: CoinType) acquires Balance
+```
+
+It is responsible for the transfer `from` the account from to the address `to`, and the transfer amount is `amount`.
+
+Let's ignore the abort condition for now and only consider its functional nature, and try to write its validation specification:
+
+```rust
+spec transfer {
+    let addr_from = Signer::address_of(from);
+
+    let balance_from = global<Balance<CoinType>>(addr_from).coin.value;
+    let balance_to = global<Balance<CoinType>>(to).coin.value;
+    let post balance_from_post = global<Balance<CoinType>>(addr_from).coin.value;
+    let post balance_to_post = global<Balance<CoinType>>(to).coin.value;
+
+    ensures balance_from_post == balance_from - amount;
+    ensures balance_to_post == balance_to + amount;
+}
+```
+
+Here `from` is of type `signer`, not a direct address.
+Although in the program we have created a local variable called `addr_from`, we cannot directly reference it in the spec block.
+At the same time, the expression of this address needs to be repeated several times, and it is very cumbersome to write repeatedly. We bind it to `addr_from` again.
+Then use `let` and `let post` to define several variables, corresponding to the balances in the two addresses `addr_from` and `to` before and after the function is executed.
+Finally, use the `ensures` statement to tell Prover that the balance in `from` should be subtracted by `amount`; the balance in `to` should be increased by `amount`.
+
+At first glance, there seems to be no problem at all. But is it really so?
+Let's see if Prover thinks this is "the correct description of the behavior of this function".
+After typing `mpm package prove` you can see:
+
+```
+error: post-condition does not hold
+   ┌─ ./sources/BasicCoin.move:58:9
+   │
+58 │         ensures balance_from_post == balance_from - amount;
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │
+   =     at ./sources/BasicCoin.move:45: transfer
+   =     at ./sources/BasicCoin.move:51: transfer (spec)
+   =     at ./sources/BasicCoin.move:53: transfer (spec)
+   =     at ./sources/BasicCoin.move:54: transfer (spec)
+   =     at ./sources/BasicCoin.move:45: transfer
+   =         from = signer{0x0}
+   =         to = 0x0
+   =         amount = 1
+   =         _witness = <generic>
+```
+
+It is unexpectedly unexpected. Prover prompts that the rear conditions are not satisfied, indicating that the behavior described in the previous spec block is not exactly the same as the `transfer` function.
```

Comment:
"unexpectedly unexpected" is a repetition. I suggest change it to "It is somewhat out of our expectation"